src/minefield.rs

```diff
@@ -144,15 +157,65 @@ impl Board {
     /// Uncovering every non-mine tile is the win condition.
     /// Note that the mine tiles are **not** required to be flagged (looking at you, speedrunners).
     pub fn handle_uncover(&mut self, x: usize, y: usize) {
-        todo!()
+        let tile_idx = self.coords_to_index(x, y);
+
+        if !self.placed {
+            self.placed = true;
+            self.place_mines(&[tile_idx]);
```

Comment:
This should also include the neighboring tiles

Reply:
What do you mean by that? To ensure that there is a square around it?

Reply:
yup, otherwise you can get cornered on the first move 😆

Reply:
New patch with improvement regarding this comment :)